autokeras/text/text_supervised.py

```diff
@@ -36,108 +36,50 @@ class TextSupervised(SingleModelSupervised, ABC):
 
     def __init__(self, verbose, **kwargs):
         super().__init__(verbose=verbose, **kwargs)
-        self.device = Backend.get_device()
-
-        # BERT specific
-        self.bert_model = 'bert-base-uncased'
-        self.tokenizer = BertTokenizer.from_pretrained(self.bert_model, do_lower_case=True)
-
-        # Labels/classes
-        self.num_labels = None
-
-    @abstractmethod
-    def fit(self, x, y, time_limit=None):
-        pass
-
-    @abstractmethod
-    def predict(self, x_test):
-        pass
-
-    @property
-    @abstractmethod
-    def metric(self):
-        pass
-
-    @property
-    @abstractmethod
-    def loss(self):
-        pass
-
-    @abstractmethod
-    def preprocess(self, x):
-        pass
-
-    def transform_y(self, y):
-        pass
-
-    def inverse_transform_y(self, output):
-        return np.argmax(output, axis=1)
-
-
-class TextRegressor(TextSupervised):
-    pass
-
-
-class TextClassifier(TextSupervised):
-    """A TextClassifier class based on Google AI's BERT model.
-
-    Attributes:
-        device: Specific hardware for using/running the model. E.g:- CPU, GPU or TPU.
-        verbose: Mode of verbosity.
-        bert_model: Type of BERT model to be used for the classification task. E.g:- Uncased, Cased, etc.
-        tokenizer: Tokenizer used with BERT model.
-        num_labels: Number of output labels for the classification task.
-        output_model_file: File location to save the trained model.
-    """
-
-    def __init__(self, verbose, **kwargs):
-        """Initialize the TextClassifier.
-
-        Args:
-            verbose: Mode of verbosity.
-        """
-        super().__init__(verbose=verbose, **kwargs)
-        self.device = Backend.get_device()
-        self.verbose = verbose
+        self.device = get_device()
 
         # BERT specific
         self.bert_model = 'bert-base-uncased'
         self.max_seq_length = 128
         self.tokenizer = BertTokenizer.from_pretrained(self.bert_model, do_lower_case=True)
 
-        # Labels/classes
+        # Evaluation params
+        self.eval_batch_size = 32
+
+        # Number of labels.
         self.num_labels = None
 
-        # Output directory
-        self.output_model_file = os.path.join(self.path, 'pytorch_model.bin')
+        # Output model directory
+        self.output_model_file = None
 
-        # Evaluation params
-        self.eval_batch_size = 32
+        # Task type - Regression or Classification.
+        self.task_type = None
```

Comment:
I think we don't need this variable.
It should be done by different subclasses.